aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -544,18 +576,35 @@ void flip_kernel(TensorIterator& iter, const bool quantized) {
       auto iter_dtype = iter.dtype();
       if (iter_dtype == kByte) {
         return cpu_hflip_vec<uint8_t>(iter);
-      } else if (iter_dtype == kFloat) {
-        return cpu_hflip_vec<float>(iter);
-      } else if (iter_dtype == kInt) {
-        return cpu_hflip_vec<int32_t>(iter);
       } else if (iter_dtype == kShort) {
         return cpu_hflip_vec<int16_t>(iter);
+      } else if (iter_dtype == kInt) {
+        return cpu_hflip_vec<int32_t>(iter);
       } else if (iter_dtype == kLong) {
         return cpu_hflip_vec<int64_t>(iter);
+      } else if (iter_dtype == kFloat) {
+        return cpu_hflip_vec<float>(iter);
       } else if (iter_dtype == kDouble) {
         return cpu_hflip_vec<double>(iter);
       }
       // other dtypes are handled below with cpu_kernel_vec
+    } else if (iter.ndim() > 1 && output_strides[1] < 0 && input_strides[0] == iter.element_size(1) \
+               && output_strides[0] == iter.element_size(0)) {
+      auto iter_dtype = iter.dtype();
+      if (iter_dtype == kByte) {
```

Comment:
shall we use `AT_DISPATCH_ALL_TYPES` here to simplify the code ? Guess only `kChar` is missing here .

And i saw you add the `inline Vectorized<T> flip(const Vectorized<T> & data)` in the vectorized utils, and the specification is made on `float`, `double`, `int8_t`, `int16_t`, `int32_t`, `int64_t`. But `uint8_t` is missing, this will probably go to the base type, please double check.

Reply:
@mingfeima thanks a lot for pointing out about non-implemented `Vectorized<uint8_t>` and missing flip for uint8 ! I used a temporary code in the latest commit before we merge https://github.com/pytorch/pytorch/pull/89284 and torch flip is now faster than Pillow on uint8 which would be helpful for vision data aug pipelines.